File: 09-mosquitto-nodejs/sub.js
const client = require('mqtt').connect(process.env.MQTT_HOST || 'mqtt://localhost:1883');

client.on('connect', () => {
    console.log("Connected to MQTT broker");
    client.subscribe("sensoren/+/temp", (err) => {
        if(err) console.err(err);
    });
})

client.on('message', (topic, message) => {
    console.log(topic, message.toString());
});

File: 10-mosquitto-python/sub.py
import paho.mqtt.client as mqtt
import os

def on_connect(client, userdata, flags, rc):
    client.subscribe("test/rnd")
    if rc != 0:
        print("Error while subscribing to test")

def on_message(client, userdata, msg):
    print(msg.topic, msg.payload.decode())

client = mqtt.Client()
client.on_connect = on_connect
client.on_message = on_message
client.connect(os.environ.get('MQTT_HOST', 'localhost'), 1883)

try:
    client.loop_forever()
except KeyboardInterrupt:
    print("Beende Programm...")
    client.loop_stop()
    client.disconnect()